update release notes for ff.com for 1.0 (#10805)
microsoft/FluidFramework: docs/content/docs/updates/v1.0.0.md

---
title: Fluid Framework 1.0
aliases:
  - /updates/1.0/
summary: |
  The 1.0 release is the first major release of the Fluid Framework.
---

## Patch Updates

### Update 1.0.1:
- Add the missing `ConnectionState` export to `fluid-framework`
- Fix AzureClient issue where the second user could not connect in `local` mode

## Breaking changes

### connected() on FluidContainer is replaced with connectionState

The `connected()` method has been replaced with the `connectionState` property. Read more

```js
import { ConnectionState } from "fluid-framework";

// Previous Syntax
if (container.connected) {
// New Syntax
if (container.connectionState === ConnectionState.Connected) {
  console.log("Container is connected");
}
```

### AzureClient connection configuration has distinct local and remote connection strings

> The AzureClient connection config has been updated to have distinct types to differentiate between local and remote connections. There is now an `AzureRemoteConnectionConfig` and `AzureLocalConnectionConfig` type that is strongly typed to the requirements of each connection. You will notice that since the `tenantID` is not need in the local case it is not longer required on the config.

Reviewer: ```suggestion
The AzureClient connection config has been updated to have distinct types to differentiate between local and remote connections. There is now an `AzureRemoteConnectionConfig` and `AzureLocalConnectionConfig` type that is strongly typed to the requirements of each connection. You will notice that since the `tenantID` is not needed in the local case, it is no longer required on the config.
```

  tenantId = "AZURE_TENANT_ID",
  //...
}
// Previous Local Syntax
import { LOCAL_MODE_TENANT_ID } from "@fluidframework/azure-client";
const connection = {
  tenantId = LOCAL_MODE_TENANT_ID,
  //...
}
```

```js
// New Remote Syntax
import { AzureRemoteConnectionConfig } from "@fluidframework/azure-client";
const connection: AzureRemoteConnectionConfig = {
  type= "remote",
  tenantId= "AZURE_TENANT_ID",
  //...
}
// New Local Syntax
import { AzureLocalConnectionConfig } from "@fluidframework/azure-client";
const connection: AzureLocalConnectionConfig = {
  type= "local",
  //...
}
```

### AzureClient connection configuration has distinct unified endpoint

The AzureClient connection config has been changed to have a single endpoint instead of the orderer and storage endpoints. This change allows us to enable multiple region support for Azure Fluid Relay as well as Disaster Recovery (DR).

```js
// Previous Syntax
const connection = {
  orderer= "AZURE_ORDERER_ENDPOINT"
  storage= "AZURE_STORAGE_ENDPOINT"
  //...
}
```

```js
// New Syntax
const connection = {
  endpoint= "AZURE_ENDPOINT"
}
```

## Other notable changes

### AzureClient supports corrupted container data recovery

> We've added a two new methods to AzureClient that will enable developers to recover data from corrupted containers. The Fluid Framework automatically generates and saves snapshots of the operation stream that we use to load the latest container. `getContainerVersions` will allow developers to access previous versions of the container. `copyContainer` allows developers to generate a new detached container from a another container.

Reviewer: ```suggestion
We've added two new methods to AzureClient that will enable developers to recover data from corrupted containers. The Fluid Framework automatically generates and saves snapshots of the operation stream that we use to load the latest container. `getContainerVersions` will allow developers to access previous versions of the container. `copyContainer` allows developers to generate a new detached container from another container.
```

---
title: Fluid Framework 1.0
aliases:
  - /updates/1.0/
summary: |
  The 1.0 release is the first major release of the Fluid Framework.
---

## Breaking changes

> ### connected() on FluidContainer is replaced with connectionState

Reviewer 2: ```suggestion
### connected on FluidContainer is replaced with connectionState
```

> The `connected()` method has been replaced with the `connectionState` property. Read more

Author: @scottn12 , can you validate this section and make updates as you see fit.

> The `connected()` method has been replaced with the `connectionState` property. Read more

Reviewer 2: Technically this is a method since it's a getter, but it essentially acts as a boolean for end users, so I think it makes sense to call it that here.
```suggestion
The `connected` boolean has been replaced with the `connectionState` property. Read more
```

```js
// Previous Remote Syntax
const connection = {
  tenantId = "AZURE_TENANT_ID",
  //...
}
// Previous Local Syntax
import { LOCAL_MODE_TENANT_ID } from "@fluidframework/azure-client";
const connection = {
  tenantId = LOCAL_MODE_TENANT_ID,
  //...
}
```

```js
// New Remote Syntax
import { AzureRemoteConnectionConfig } from "@fluidframework/azure-client";
const connection: AzureRemoteConnectionConfig = {
  type= "remote",
  tenantId= "AZURE_TENANT_ID",
  //...
}
// New Local Syntax
import { AzureLocalConnectionConfig } from "@fluidframework/azure-client";
const connection: AzureLocalConnectionConfig = {
  type= "local",
  //...
}
```

### AzureClient connection configuration has distinct unified endpoint

The AzureClient connection config has been changed to have a single endpoint instead of the orderer and storage endpoints. This change allows us to enable multiple region support for Azure Fluid Relay as well as Disaster Recovery (DR).

```js
// Previous Syntax
const connection = {
  orderer= "AZURE_ORDERER_ENDPOINT"
  storage= "AZURE_STORAGE_ENDPOINT"
  //...
}
```

```js
// New Syntax
const connection = {
  endpoint= "AZURE_ENDPOINT"
}
```

## Other notable changes

> ### AzureClient supports corrupted container data recovery

Reviewer 3: Looks concise! Minor suggestion there.

We've added a two new methods to AzureClient that will enable developers to recover data from corrupted containers. The Fluid Framework automatically generates and saves snapshots of the operation stream that we use to load the latest container. `getContainerVersions` will allow developers to access previous versions of the container. `copyContainer` allows developers to generate a new detached container from a another container.

[`getContainerVersions(id, options)`](https://fluidframework.com/docs/apis/azure-client/azureclient/#azure-client-azureclient-getcontainerversions-Method)
[copyContainer(id, containerSchema)](https://fluidframework.com/docs/apis/azure-client/azureclient/#azure-client-azureclient-copycontainer-Method)

In an situation where a container will fail to load these two methods can be used together to load a document from a previous state in time.

### AzureClient now supports `getContainerVersions`

> `getContainerVersions` on the AzureClient allows developers to view the previously generated snapshot states of the Container.

Reviewer 3: ```suggestion
`getContainerVersions` on the AzureClient allows developers to view the previously generated snapshot versions of the Container.
```
